chore(customrawdb): delete customrawdb package and switch to avalanchego imports

powerslider opened the pull request:
## Why this should be merged

Check ava-labs/avalanchego#4569 

## How this works

- Delete `plugin/evm/customrawdb`.
- Switch dependent code to using `github.com/ava-labs/avalanchego/vms/evm/sync/customrawdb` import instead.

## How this was tested

successful build

## Need to be documented?

no

## Need to update RELEASES.md?

no 

resolves ava-labs/avalanchego#4569

Signed-off-by: Tsvetan Dimitrov (tsvetan.dimitrov23@gmail.com)

powerslider committed:
chore(customrawdb): delete customrawdb package and switch to avalanchego imports

resolves #1302

Signed-off-by: Tsvetan Dimitrov (tsvetan.dimitrov23@gmail.com)

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
fix: rlpgen import issue and update to latest avalanchego version

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
ci: update ci.yml with the updated avalanchego version

JonathanOppenheimer commented on plugin/evm/customtypes/hashing_test.go:
```diff
@@ -43,6 +42,8 @@ import (
 	"github.com/ava-labs/libevm/trie"
 	"github.com/ava-labs/libevm/triedb"
+
+	mrand "math/rand"
```
Is there a reason for this change?

powerslider replied:
Yep, `goimports` or `gci` lint error.

JonathanOppenheimer replied:
You should just not alias it -- there's no reason to.

JonathanOppenheimer reviewed (dismissed):
Difference in this repository looks good - IMO, should not be merged until there's master commit that can be referenced.

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
chore: update to latest avalanchego version

powerslider committed:
fix: add input and output sorting for the comm command

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
fix: change customrawdb calls in test_syncervm

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
chore(go.mod) upgrade avalanchego to 3b08d6131a00bd55349321a58ae7e6cfcff8635e

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

Signed-off-by: Tsvetan Dimitrov <tsvetan.dimitrov23@gmail.com>

powerslider committed:
fix(statesync): handle ErrNotFound when reading sync root

Handle database.ErrNotFound returned by ReadSyncRoot when there's no
previous sync root. This is expected for fresh databases and should be
treated as an empty hash (no previous sync) rather than an error.

powerslider committed:
fix(vmtest): improve requireSyncPerformedHeight

powerslider committed:
fix(statesync): handle ErrNotFound when reading sync root

powerslider committed:
fix(core): handle ErrNotFound when reading acceptor tip

Handle database.ErrNotFound from ReadAcceptorTip as expected case
(empty hash) instead of error. This fixes VM initialization failures
on fresh databases.

powerslider committed:
fix(snapshot): handle ErrNotFound in wipe test

Update TestWipe to accept database.ErrNotFound as valid result
after wipe, since ReadSnapshotBlockHash returns ErrNotFound when
the marker doesn't exist.

powerslider committed:
ci: upgrade ci.yml avalanchego versions to correspond with go.mod change

powerslider committed:
fix(tracers): apply block override before creating block context

- Move block number override fix before NewEVMBlockContext creation
so GetHashFn can correctly resolve blockhash(n) when block number
is overridden to n+1. This fixes TestTracingWithOverrides for
Firewood scheme.

- The bug was probably a pre-existing code issue where the header modification
happened after creating the block context. Since GetHashFn creates
a closure that captures the header reference, modifying the header
afterward left the closure with stale values. This bug was exposed
by the recent avalanchego update (3b08d6131a00bd55349321a58ae7e6cfcff8635e),
which likely changed behavior that made the test fail specifically
for Firewood scheme, while HashScheme continued to pass.

JonathanOppenheimer commented on scripts/build_test.sh:
```diff
@@ -20,4 +20,5 @@ if [[ -n "${NO_RACE:-}" ]]; then
 # shellcheck disable=SC2046
-go test -shuffle=on ${race:-} -timeout="${TIMEOUT:-600s}" -coverprofile=coverage.out -covermode=atomic "$@" $(go list ./... | grep -v github.com/ava-labs/coreth/tests) 
+go test -shuffle=on ${race:-} -timeout="${TIMEOUT:-600s}" -coverprofile=coverage.out -covermode=atomic "$@" $(go list ./... | grep -v github.com/ava-labs/coreth/tests)
+
```
is this two empty lines at the bottom of the file?

JonathanOppenheimer reviewed (changes requested):
See comments

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
fix(scripts): remove unnecessary empty line

powerslider committed:
refactor(snapshot): improve error handling and test validation

- Extract sentinel errors in journal.go for better error checking
and maintainability.
- Refactor loadSnapshot to use switch statements
for clearer control flow when validating snapshot block hash and root.
- Improve TestWipe validation logic:
   - Use switch statements for consistent error handling patterns.
   - Add context to error messages ("before wipe" / "after wipe").

powerslider committed:
fix(customtypes): unalias math/rand package

alarso16 commented on plugin/evm/vmtest/test_syncervm.go:
```diff
@@ -653,16 +654,17 @@ func generateAndAcceptBlocks(t *testing.T, vm extension.InnerVM, numBlocks int,
+// requireSyncPerformedHeight verifies the latest sync performed height matches expectations.
+// If `expected` is nil, verifies the result is 0 (no sync performed).
+// If `expected` is non-nil, verifies the result equals the expected height.
+func requireSyncPerformedHeight(t *testing.T, db ethdb.KeyValueStore, expected *uint64) {
```
Why is this a pointer? It just seems like you could send an integer

powerslider commented on core/state/snapshot/wipe_test.go:
The changes were needed, because the test broke, but I also addressed some stylistic comments from @JonathanOppenheimer

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package

powerslider committed:
refactor(vmtest): simplify requireSyncPerformedHeight signature

ceyonur commented on eth/tracers/api.go:
```diff
@@ -961,21 +961,23 @@ func (api *API) TraceCall(ctx context.Context, args ethapi.TransactionArgs, bloc
 	h := block.Header()
+	// Apply block number override fix before creating block context so GetHashFn
+	// can correctly resolve blockhash(n) when block number is overridden to n+1.
+	if config != nil && config.BlockOverrides != nil && config.BlockOverrides.Number.ToInt().Uint64() == h.Number.Uint64()+1 {
```
I'd defer this to @ARR4N as he put up the fix.

ARR4N replied:
> I'd defer this to [@ARR4N](https://github.com/ARR4N) as he put up the fix.

This was a pseudo-cherry-pick (#1301) from an upstream security fix.

powerslider committed:
Merge branch 'master' into powerslider/1302-delete-customrawdb-package
